packages/client/src/components/app/forms/RelationshipField.svelte

```diff
@@ -50,54 +36,51 @@
       limit: 100,
     },
   })
-  $: hasFilter = !!filter?.filter(f => !!f.field)?.length
-  $: fetch.update({ filter })
-  $: {
-    options = searchResults ? searchResults : $fetch.rows
-    const nonMatchingOptions = selectedOptions.filter(
-      option => !options.map(opt => opt._id).includes(option._id)
-    )
-    // Append initially selected options if there is no filter
-    // and hasn't already been appended
-    if (!hasFilter) {
-      options = [...options, ...nonMatchingOptions]
-    }
-  }
+
   $: tableDefinition = $fetch.definition
-  $: primaryDisplay = tableDefinition?.primaryDisplay || "_id"
-  $: singleValue = flatten(fieldState?.value)?.[0]
-  $: multiValue = flatten(fieldState?.value) ?? []
+  $: selectedValue = multiselect
+    ? flatten(fieldState?.value) ?? []
+    : flatten(fieldState?.value)?.[0]
   $: component = multiselect ? CoreMultiselect : CoreSelect
   $: expandedDefaultValue = expand(defaultValue)
-  $: debouncedSearch(searchString)
+  $: primaryDisplay = tableDefinition?.primaryDisplay
+
+  let optionsObj = {}
+  let initialValuesProcessed
   $: {
-    if (searching) {
-      isOpen = true
+    if (!initialValuesProcessed && primaryDisplay) {
+      initialValuesProcessed = true
+      optionsObj = {
+        ...optionsObj,
+        ...fieldState?.value?.reduce((accumulator, value) => {
+          accumulator[value._id] = {
+            _id: value._id,
+            [primaryDisplay]: value.primaryDisplay,
+          }
+          return accumulator
+        }, {}),
+      }
     }
   }
+  $: optionsObj = {
+    ...optionsObj,
```

Comment:
Does this work alright... it's basically a recursive declaration but I guess svelte can work out how to handle it?

It might be a bit clearer to refactor this to have an `optionsObj` as you have, then something like
```js
$: enrichedOptions = enrichOptions(optionsObj, $fetch.rows)
```
and break the logic out into a function.

I just think the recursive declaration is a little confusing, especially in future whenever the next dev comes to work on this file.

Or maybe just renaming them would be enough so that the reactive part is actually a new variable with a different name, rather than reactively updating a variable with the same name.